include/RAJA/MultiPolicy.hxx

```diff
@@ -0,0 +1,135 @@
+ *
+ ******************************************************************************
+ */
+
+#ifndef RAJA_MultiPolicy_HXX
+#define RAJA_MultiPolicy_HXX
+
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016, Lawrence Livermore National Security, LLC.
+//
+// Produced at the Lawrence Livermore National Laboratory
+//
+// LLNL-CODE-689114
+//
+// All rights reserved.
+//
+// This file is part of RAJA.
+//
+// For additional details, please also read RAJA/LICENSE.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice,
+//   this list of conditions and the disclaimer below.
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the disclaimer (as noted below) in the
+//   documentation and/or other materials provided with the distribution.
+//
+// * Neither the name of the LLNS/LLNL nor the names of its contributors may
+//   be used to endorse or promote products derived from this software without
+//   specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL LAWRENCE LIVERMORE NATIONAL SECURITY,
+// LLC, THE U.S. DEPARTMENT OF ENERGY OR CONTRIBUTORS BE LIABLE FOR ANY
+// DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+// OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+// HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+// IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "RAJA/RAJA.hxx"
+
+#include <tuple>
+
+namespace RAJA
+{
+
+template <typename Selector, typename ...Policies>
+class MultiPolicy {
+  Selector s;
+public:
+  MultiPolicy() = delete; // No default construction
+  MultiPolicy(Selector s) : s(s)
+  { }
+
+  MultiPolicy(const MultiPolicy &p) : s(p.s)
+  { }
+
+  template <typename Iterable>
+  int operator()(Iterable &&i)
+  {
+    return s(i);
+  }
+};
+
+template <typename ...Policies, typename Selector>
+auto make_multi_policy(Selector s) -> MultiPolicy<Selector, Policies...>
+{
+  return MultiPolicy<Selector, Policies...>(s);
+}
+
+namespace detail
+{
+
+template <size_t index, size_t size, typename first, typename... rest>
+struct policy_invoker {
+  using policy = first;
```

Comment:
I would add a `static_assert` here in case the user attempts to call the multi-policy again. This would clean up the ~awful~ *awesome* error message that would most likely be emitted.